domain/application/state/charm.go

```diff
@@ -302,29 +302,35 @@ func (s *CharmState) ReserveCharmRevision(ctx context.Context, id corecharm.ID,
 		return "", errors.Trace(err)
 	}
 
-	var charmResult charmIDName
+	var reserveCharmResult reserveCharm
 	ident := charmID{UUID: id.String()}
 
 	selectQuery := `
-SELECT &charmIDName.*
-FROM charm 
-LEFT JOIN charm_state
-ON charm.uuid = charm_state.charm_uuid
+SELECT &reserveCharm.*
+FROM charm
+JOIN charm_metadata AS cm ON charm.uuid = cm.charm_uuid
+JOIN charm_origin AS co ON charm.uuid = co.charm_uuid
 WHERE uuid = $charmID.uuid;
 `
-	selectStmt, err := s.Prepare(selectQuery, charmResult, ident)
+	selectStmt, err := s.Prepare(selectQuery, ident, reserveCharmResult)
+	if err != nil {
+		return "", fmt.Errorf("failed to prepare query: %w", err)
```

Comment:
I think we are starting to use (juju/internal)`errors.Errorf(` instead of the fmt one.

Reply:
I'll make a PR to fix them all in the future. This way I have a clear focused PR.